content/docs/start/model-management/model-cicd.md

```diff
@@ -0,0 +1,234 @@
+  run: |
+    dvc config --global studio.token ${{ secrets.DVC_STUDIO_TOKEN }}
+    dvc artifacts get  ${{ github.server_url }}/${{ github.repository }} ${{ needs.parse.outputs.name }} --rev ${{ needs.parse.outputs.version }}
+    echo "The right model is available and you can use the rest of this command to deploy it. Good job!"
+```
+
+Here, we are using the outputs of the `parse` job to specify the correct model
+version. We are then setting up the DVC Studio token which we stored in our
+GitHub repository as a
+[secret](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions)
+to manage authentication with the
+[DVC remote storage](https://dvc.org/doc/user-guide/data-management/remote-storage#remote-storage).
+This way we only need to keep the Studio token saved on GitHub and let Studio
+manage the specific storage credentials for us.
+
+Finally, `github.server_url` and `github.repository` are
+[default environmental variables in GitHub](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)
+which together form the URL of our repository on GitHub. We could of course also
+specify the URL manually.
+
+If you don't use Studio, you can still use `dvc artifacts get` but you will need
+to keep your remote storage credentials on GitHub and use them to configure DVC
+in the CICD workflow. You will also need to checkout the repository in the
+workflow. You can see more details in the
+[documentation](/doc/command-reference/artifacts/get#description).
+
+You can now use the following template to create your own Model Registry CICD
+actions on GitHub!
+
+```yaml
+name: Template CICD Action for DVC Model Registry
+
+on:
+  # the workflow is triggered whenever a tag is pushed to the repository
+  push:
+    tags:
+      - "*"
+
+jobs:
+
+  # This job parses the git tag with the GTO GitHub Action to identify model registry actions
+  parse:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: "Parse GTO tag"
+      id: gto
+      uses: iterative/gto-action@v2
+    outputs:
+      event: ${{ steps.gto.outputs.event }}
+      name: ${{ steps.gto.outputs.name }}
+      stage: ${{ steps.gto.outputs.stage }}
+      version: ${{ steps.gto.outputs.version }}
+
+  perform-action-with-model:
+    needs: parse
+    if: # here paste the model registry event condition you want to trigger your action
+    steps:
+    - uses: iterative/setup-dvc@v1
+        # this step uses DVC to download the model artifact from our remote repository so we can perform CICD actions with it
+        # The DVC Studio token is used to avoid having to store specific remote storage credentials on GitHub
+        - name: Get Model For Deployment
+          run: |
+            dvc config --global studio.token ${{ secrets.DVC_STUDIO_TOKEN }}
+            dvc artifacts get  ${{ github.server_url }}/${{ github.repository }} ${{ needs.parse.outputs.name }} --rev ${{ needs.parse.outputs.version }}
```

Comment:
Not a blocker, but should we update to reflect the current workflow, including filtering to prod stage assignments only?

Reply:
Well, my idea behind this is that you can copy-paste this as a template. and you might use different conditions or want to do different things with the second job. But you will still probably want to do something with your model. But I don't mind either way...do you think it is confusing like this @dberenbaum ?

Reply:
I think it's slightly confusing, and it's easier to modify the filter than to have to write it from scratch (maybe a comment could help?).

Reply:
Ok, let's do it that way